Documentation/specs/runtime-configuration-file.md

```diff
@@ -0,0 +1,267 @@
+
+The exact settings found here are specific to the compiler that produced the original application binary. Some example settings include: `defines`, `languageVersion` (C#/VB), `allowUnsafe` (C#), etc.
+
+As an example, here is a possible `project.json` file:
+
+```json
+{
+    "compilationOptions": {
+        "allowUnsafe": true
+    },
+    
+    "frameworks": {
+        "net451": {
+            "compilationOptions": {
+                "defines": [ "DESKTOP_CLR" ]
+            }
+        },
+        "dnxcore50": {
+            "compilationOptions": {
+                "defines": [ "CORE_CLR" ]
+            }
+        }
+    },
+    
+    "configurations": {
+        "Debug": {
+            "compilationOptions": {
+                "defines": [ "DEBUG_MODE" ]
+            }
+        }
+    }
+}
+```
+
+When this project is built for `dnxcore50` in the `Debug` configuration, the outputted `runtime.config.json` file will have the following `compilationOptions` section:
+
+```json
+{
+    "compilationOptions": {
+        "allowUnsafe": true,
+        "defines": [ "CORE_CLR", "DEBUG_MODE" ]
+    }
+}
+```
+
+### `targets` Section
+
+This section contains subsetted data from the input `project.lock.json`.
+
+Each property under `targets` describes a "target", which is a collection of libraries required by the application when run or compiled in a certain framework and platform context. A target **must** specify a Framework name, and **may** specify a Runtime Identifier. Targets without Runtime Identifiers represent the dependencies and assets used for compiling the application for a particular framework. Targets with Runtime Identifiers represent the dependencies and assets used for running the application under a particular framework and on the platform defined by the Runtime Identifier. In the example above, the `DNXCore,Version=v5.0` target lists the dependencies and assets used to compile the application for `dnxcore50`, and the `DNXCore,Version=v5.0/osx.10.10-x64` target lists the dependencies and assets used to run the application on `dnxcore50` on a 64-bit Mac OS X 10.10 machine.
+
+There will always be two targets in the `runtime.config.json` file: A compilation target, and a runtime target. The compilation target will be named with the framework name used for the compilation (`DNXCore,Version=v5.0` in the example above). The runtime target will be named with the framework name and runtime identifier used to execute the application (`DNXCore,Version=v5.0/osx.10.10-x64` in the example above). **Open Question**: We probably don't want `corehost` to have to select or even parse target names. Maybe we should have a separate entry that provides the exact names so that the host can just read that and then match it against the targets?
```

Comment:
I concur with the proposal of the open question. The host should not be implementing policy of parsing - otherwise if that changes in future, it will require host to be updated as well.